Sources/Sentry/SentryThreadHandle.cpp

```diff
@@ -74,6 +77,10 @@ namespace profiling {
         mach_msg_type_number_t count;
         thread_act_array_t list;
         auto current = ThreadHandle::current();
+        if (current == nullptr) {
+            return std::make_pair(threads, nullptr);
```

Comment:
I just went back to read about [std::move](https://en.cppreference.com/w/cpp/utility/move) again and still don't understand what it's meant for XD Please let us know @indragiek .

Reply:
I think this might also be causing this error which is breaking CI:

> /Applications/Xcode15.1/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.2.sdk/usr/include/c++/v1/__memory/allocator.h:168:28 Call to implicitly-deleted copy constructor of 'std::unique_ptr<sentry::profiling::ThreadHandle>'

Reply:
Back on the question about `std::move`, I just came across this SO answer that mentions it wrt `std::unique_ptr`: https://stackoverflow.com/a/6876833, that mentions using std::move when you need to create a subsequent reference to an object managed by a `unique_ptr`, which can only ever have one reference to it. Is there another use for std::move, or is this the gist of it @indragiek ? In that case, I think we don't need std::move here, because `threads` is a vector of `unique_ptr`, not a `unique_ptr` itself. Looking for confirmation on that though.

Reply:
It's not needed here:`std::move` on the vector would move the vector into the `std::pair` instead of making a copy of the vector and then storing the copy in the pair, but since the vector is empty here it doesn't make any difference

On the usage of `std::move` line 98 the vector has elements in it so we do avoid copying a vector by moving it instead

Reply:
Makes sense, thanks!